third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -79,13 +81,26 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   std::vector<ComputationClient::DataPtr> datas;
   datas.reserve(tensors.size());
   for (auto& tensor : tensors) {
-    xla::Literal literal(tensor.shape);
-    tensor.populate_fn(tensor, literal.untyped_data(), literal.size_bytes());
-
     PjRtDevice* pjrt_device = StringToPjRtDevice(tensor.device);
-    std::shared_ptr<xla::PjRtBuffer> buffer =
-        client_->BufferFromHostLiteral(literal, pjrt_device).ValueOrDie();
-    buffer->GetReadyFuture().Await();
+
+    auto literal = std::make_shared<xla::Literal>(tensor.shape);
+    tensor.populate_fn(tensor, literal->untyped_data(), literal->size_bytes());
+    absl::InlinedVector<int64_t, 4> byte_strides(
+        literal->shape().dimensions_size());
+    ShapeUtil::ByteStrides(literal->shape(), absl::MakeSpan(byte_strides));
+
+    // Avoid use-after-free on `literal` due to unsequenced move and use.
+    xla::Literal* literal_pointer = literal.get();
+    std::shared_ptr<xla::PjRtBuffer> buffer = std::move(
+        client_
+            ->BufferFromHostBuffer(
```

Comment:
How do you intend to unit test this approach?

Reply:
We don't have good unit tests for either ComputationClient, unfortunately. This does get a sanity check as part of the PjRt CPU integration tests that run in the CI, and I manually ran tests on TPU.

Reply:
Sounds good. I suggest incorporating unit tests for the main building blocks of PjRt integration in a future set of PRs.

Reply:
Agreed. Filed a cleanup ticket internally.